nixos/modules/services/audio/mpdscribble.nix

```diff
@@ -0,0 +1,242 @@
+          type = type.string;
+          description = ''
+            The handshake URL of the scrobbler.
+          '';
+        };
+
+        username = mkOption {
+          default = "";
+          example = "";
+          type = type.string;
+          description = ''
+            Last.fm User
+          '';
+        };
+
+        password = mkOption {
+          default = "";
+          example = "";
+          type = type.str;
+          description = ''
+            Last.fm Password
+
+            You should really not specify your plain-text password here, but
+            pass the MD5 hash of it, as mpdscribble suggests.
+
+            For further information look here: http://dev.man-online.org/man1/mpdscribble/
+            or in your local copy of the manpage.
+          '';
+        };
+
+        journal = mkOption {
+          default = "";
+          example = /home/user/.mpdscribble.lastfm.cache;
+          type = type.path;
+          description = ''
+            (Optional) file path for caching scrobbles when no internet connection
+            is available.
+          '';
+        };
+
+      };
+
+      librefm = {
+
+        enable = mkOption {
+          default = false;
+          description = ''
+            Enable Librefm scrobbling.
+          '';
+        };
+
+        url = mkOption {
+          default = ''http://turtle.libre.fm/'';
+          example = ''http://turtle.libre.fm/'';
+          type = type.string;
+          description = ''
+            The handshake URL of the scrobbler.
+          '';
+        };
+
+        username = mkOption {
+          default = "";
+          example = "";
+          type = type.string;
+          description = ''
+            Librefm user.
+          '';
+        };
+
+        password = mkOption {
+          default = "";
+          example = "";
+          type = type.string;
+          description = ''
+            Librefm Password
+
+            You should really not specify your plain-text password here, but
+            pass the MD5 hash of it, as mpdscribble suggests.
+
+            For further information look here: http://dev.man-online.org/man1/mpdscribble/
+            or in your local copy of the manpage.
+          '';
+        };
+
+        journal = mkOption {
+          default = "";
+          example = /home/user/.mpdscribble.librefm.cache;
+          type = type.path;
+          description = ''
+            (Optional) file path for caching scrobbles when no internet connection
+            is available.
+          '';
+        };
+
+      };
+
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.mpdscribble = {
+      after         = [ "mpd.service" ];
+      description   = "MPD scrobble client for lastfm/librefm";
+      wants         = [ "mpd.service" ];
+      serviceConfig = {
+        User = cfg.user;
+        Group = cfg.group;
+        ExecStart = "${pkgs.mpdscribble}/bin/mpdscribble --no-daemon";
```

Comment:
It occurred to me that you need to add `--conf ${mpdscribbleConf}` as well, otherwise it will try to load mpdscribble.conf from its homedir.